### PR TITLE
Add search field component

### LINKED ISSUE #1146: [Search Field] Development

### PR DESCRIPTION
**Pull request by jpzwarte**
Fixes #1146

### CODE REVIEW

File: packages/components/search-field/src/search-field.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  /** @internal Emits when the user presses enter. */
+  @event({ name: 'sl-search' }) searchEvent!: EventEmitter<SlSearchEvent>;
+
+  override renderPrefix(): TemplateResult {
+    return html`<sl-icon name="search"></sl-icon>`;
+  }
+
+  override renderSuffix(): TemplateResult | typeof nothing {
+    return this.value && !this.disabled
+      ? html`<button @click=${this.#onClick}><sl-icon name="xmark"></sl-icon></button>`
```
Comment (anna-lach) [inline packages/components/search-field/src/search-field.ts:45]: Not `sl-button`, just a `button`?

File: packages/components/search-field/package.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
+{
+  "name": "@sl-design-system/search-field",
+  "version": "0.0.0",
+  "description": "Text field component for the SL Design System",
```
Comment (Diaan) [inline packages/components/search-field/package.json:4]:
```suggestion
  "description": "Search field component for the SL Design System",
```

File: packages/components/search-field/src/search-field.scss [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+button {
+  align-items: center;
+  appearance: none;
+  background: transparent;
+  border: 0;
+  cursor: pointer;
+  display: inline-flex;
+  padding: var(--_button-padding);
+}
```
Comment (Diaan) [inline packages/components/search-field/src/search-field.scss:15]: This button is able to get the focus when using keyboard navigation, but it doesn't use the default focus ring (styling), causing it to look like this in Max for example:
image

Comment (Diaan) [reply packages/components/search-field/src/search-field.scss:15]: There is of course also a case to be made for it not being focusable with the keyboard. (although I'm not sure if the 'esc' functionality is commonly known)

**Review by Diaan — CHANGES_REQUESTED** [review]

**Comment by Diaan** [comment]
In the acceptance criteria there is mention of a(n optional) button, i don't see that in the code (examples).

File: packages/components/search-field/src/search-field.ts (hunk 1 above)
Comment (jpzwarte) [reply packages/components/search-field/src/search-field.ts]: I've changed it to a small `<sl-button>` for now.

File: packages/components/search-field/src/search-field.scss (hunk 3 above)
Comment (jpzwarte) [reply packages/components/search-field/src/search-field.scss:7]: I've changed it to use a small `<sl-button>`. And the button is keyboard focusable.

**Review by anna-lach — APPROVED** [review]

**Pull request merged**